app/views/layouts/_main_nav.html.erb

```diff
@@ -1,26 +1,27 @@
-<% default_nav_links = NavigationLink.default_section.ordered.to_a %>
-<% other_nav_links = NavigationLink.other_section.ordered.to_a %>
+<% default_nav_ids = Rails.cache.fetch("default_navigation_links", expires_in: 15.minutes) do
+  NavigationLink.default_section.ordered.ids
+end %>
+
+<% other_nav_ids = Rails.cache.fetch("other_navigation_links", expires_in: 15.minutes) do
+  NavigationLink.other_section.ordered.ids
+end %>
```

Comment:
**Another alternative:**

Instead of having two separate caches here, one for the "default nav links" and the other for the "other nav links" we could just have one "navigation_links" cache like how I had it previously:


````
@navigation_links = Rails.cache.fetch("navigation_links", expires_in: 5.minutes) do
  {
    default_nav_links: NavigationLink.default_section.ordered.ids,
  }
end

@navigation_links = Rails.cache.fetch("navigation_links", expires_in: 5.minutes) do
  {
    other_navigation_links: NavigationLink.other_section.ordered.ids,
  }
end
```

And then in the app/controllers/admin/navigation_links_controller.rb above, when we delete the cache, we wouldn't check what type of link has been CRUD'ed, and update that specific cache, but we'd just delete the whole `navigation_links` cache with  `Rails.cache.delete("navigation_links")`.

On the surface, the way that I have it currently seem optimal because we wont be purging *all* the navigation links but we'd be purging it by the appropriate section. However, maybe that's introducing an extra dependency or not optimal in terms of spinning up two caches. I'd love more thoughts on this please?

Reply:
cc @maestromac

Reply:
Didn't mean to confuse you here which the example I sent you prior but yes combining them is the most optimal way.